docs/docs/components-and-props.md

```diff
@@ -0,0 +1,247 @@
+---
+id: components-and-props
+title: Components and Props
+permalink: docs/components-and-props.html
+prev: rendering-elements.html
+---
+
+Components let you split the UI into independent reusable pieces, and think about each piece separately as if it existed in the isolation.
+
+Conceptually, components are like JavaScript functions. They accept arbitrary inputs (called "props") and return React elements describing what should appear on the screen.
+
+## Functional and Class Components
+
+The simplest way to define a component is to write a JavaScript function:
+
+```js
+function Welcome(props) {
+  return <h1>Hello, {props.name}</h1>;
+}
+```
+
+This function is a valid React component because it accepts a single "props" object argument with data and returns a React element. We call such components "functional" because they are literally JavaScript functions.
+
+You can also use an [ES2015 class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes) to define a component:
+
+```js
+class Welcome extends React.Component {
+  render() {
+    return <h1>Hello, {this.props.name}</h1>;
+  }
+}
+```
+
+The above two components are equivalent from React's point of view.
+
+Classes have some additional features that we will discuss in the next sections. Until then, we will use functional components for their conciseness.
+
+## Rendering a Component
+
+Previously, we only encountered React elements that represent DOM tags:
+
+```js
+const element = <div />;
+```
+
+However, elements can also represent user-defined components:
+
+```js
+const element = <Welcome name="Sara" />;
+```
+
+When React sees an element representing a user-defined component, it passes JSX attributes to this component as a single object. We call this object "props".
```

Comment:
I think period-inside-quotes is nuts, it's just like a mistake some professor of grammar made while codifying English grammar. Let's do period-outside-quotes.